### PR TITLE
Fix issues in ENUM types and time:Utc types in entities

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes:

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/googlesheets_client.bal [hunk 1]
```diff
@@ -364,12 +364,18 @@ public isolated client class GoogleSheetsClient {
...
         } else if dataType == "decimal" {
             return decimal:fromString(value.toString());
         } else if dataType == "float" {
             return float:fromString(value.toString());
+        } else if dataType == "boolean" {
+            if value.toString() == "TRUE" {
+                return true;
+            } else {
+                return false;
+            }
```
Comment (daneshk) [inline ballerina/googlesheets_client.bal:378]:
```suggestion
            return value.toString() == "TRUE";
```

File: ballerina/tests/gsheets-all_data_types-tests.bal [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    GoogleSheetsRainierClientAllDataType rainierClientAllDataType =  check new ();
+    OrderItemExtended orderItemRetrieved = check rainierClientAllDataType->/orderitemextendeds/[orderItemExtended2.orderId]/[orderItemExtended2.itemId].put({
+        arivalTimeCivil : orderItemExtended3.arivalTimeCivil,
+        paid: orderItemExtended3.paid
+    });
+    test:assertEquals(orderItemRetrieved, orderItemExtended3Retrieved);
+
+    orderItemRetrieved = check rainierClientAllDataType->/orderitemextendeds/[orderItemExtended2.orderId]/[orderItemExtended2.itemId].get();
+    test:assertEquals(orderItemRetrieved, orderItemExtended3Retrieved);
+}
```
Comment (daneshk) [inline ballerina/tests/gsheets-all_data_types-tests.bal:67]:
```suggestion
}

```

**Review by daneshk — DISMISSED** [review]
Add minor comment. Overall LGTM

**Review by kaneeldias — APPROVED** [review]

File: ballerina/tests/init-tests.bal [hunk 3]
```diff
@@ -289,6 +331,49 @@ public type EmployeeInfo record {|
...
+    CustomerId: 1,
+    paid: false,
+    ammountPaid: 10.5f,
+    ammountPaidDecimal: 10.5,
+    arivalTimeCivil: {"timeAbbrev":"Z","dayOfWeek":5 ,"year":2024,"month":4,"day":12,"hour":12,"minute":20,"second":50.52},
+    arivalTimeUtc: [1684493685, 0.998012000],
+    arivalTimeDate: {year: 2021, month: 4, day: 12},
+    arivalTimeTimeOfDay: {hour: 17, minute: 50, second: 50.52},
+    orderType: ONLINE
+
```
Comment (kaneeldias) [inline ballerina/tests/init-tests.bal:346]:
```suggestion
```

File: ballerina/tests/test_entities_generated_types.bal [hunk 4]
```diff
@@ -254,3 +259,49 @@ public type AllTypesIdRecordUpdate record {|
     string randomField?;
 |};
 
+
```
Comment (kaneeldias) [inline ballerina/tests/test_entities_generated_types.bal:262]:
```suggestion
```

File: native/src/main/java/io/ballerina/stdlib/persist/Constants.java [hunk 5]
```diff
@@ -54,6 +54,7 @@ public static final class TimeTypes {
         public static final String CIVIL = "Civil";
         public static final String DATE_RECORD = "Date";
         public static final String TIME_RECORD = "TimeOfDay";
+        public static final String UTC = "Utc";
```
[the remark this answers is not in the document]
Comment (sahanHe) [reply native/src/main/java/io/ballerina/stdlib/persist/Constants.java]: updated in [f7d8c93](https://github.com/ballerina-platform/module-ballerina-persist/pull/218/commits/f7d8c9364ffc8fa17d1d01c211a78e2af887b9bd)

File: ballerina/tests/test_entities_generated_types.bal (hunk 4 above)
Comment (sahanHe) [reply ballerina/tests/test_entities_generated_types.bal]: updated in [f7d8c93](https://github.com/ballerina-platform/module-ballerina-persist/pull/218/commits/f7d8c9364ffc8fa17d1d01c211a78e2af887b9bd)

File: ballerina/tests/init-tests.bal (hunk 3 above)
Comment (sahanHe) [reply ballerina/tests/init-tests.bal:346]: updated in [f7d8c93](https://github.com/ballerina-platform/module-ballerina-persist/pull/218/commits/f7d8c9364ffc8fa17d1d01c211a78e2af887b9bd)

File: ballerina/tests/gsheets-all_data_types-tests.bal (hunk 2 above)
Comment (sahanHe) [reply ballerina/tests/gsheets-all_data_types-tests.bal]: updated in [f7d8c93](https://github.com/ballerina-platform/module-ballerina-persist/pull/218/commits/f7d8c9364ffc8fa17d1d01c211a78e2af887b9bd)

File: ballerina/googlesheets_client.bal (hunk 1 above)
Comment (sahanHe) [reply ballerina/googlesheets_client.bal]: updated in [f7d8c93](https://github.com/ballerina-platform/module-ballerina-persist/pull/218/commits/f7d8c9364ffc8fa17d1d01c211a78e2af887b9bd)

**Review by daneshk — APPROVED** [review]

**Pull request merged**